### PR TITLE
RPC-based Gas Estimation for Moonbeam-based Networks

### PR DESCRIPTION
**Pull request by albertov19**
## Motivation

Moonbeam introduced MBIP-5 in which gas estimation deviated from Ethereum. The reasoning is that now storage changes are reflected now on gas estimation. This is to prevent spamming the network with junk data. This is a problem on a lot of cheap-to-transact networks

## Solution

Solution was simple, just add Moonbeam-based networks to the `has_different_gas_calc` function (like with Arbitrum) [here](https://github.com/foundry-rs/foundry/compare/master...albertov19:foundry:albertov19/use-rpc-gas-estimation-moonbeam)

If this function returns `true` then gas estimation is done using the actual provider URL and `eth_estimateGas`.

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
~~the outlined solution makes sense, though I think you forgot to include it here :)~~

File: crates/cli/src/utils/cmd.rs [hunk 1]
```diff
@@ -167,11 +167,15 @@ pub fn has_different_gas_calc(chain_id: u64) -> bool {
...
-                NamedChain::ArbitrumSepolia
-        )
+            NamedChain::Arbitrum
+                | NamedChain::ArbitrumTestnet
+                | NamedChain::ArbitrumGoerli
+                | NamedChain::ArbitrumSepolia
+                | NamedChain::Moonbeam
+                | NamedChain::Moonriver
+                | NamedChain::Moonbase
+                | NamedChain::MoonbeamDev
```
Comment (mattsse) [inline crates/cli/src/utils/cmd.rs:177]: ah nvm, I overlooked it hehe

**Review by mattsse — CHANGES_REQUESTED** [review]
please restore other formatting issues like trailing ;

**Review by Evalir — CHANGES_REQUESTED** [review]
looks good, pending mattsse's nit

**Comment by albertov19** [comment]
@mattsse @Evalir thanks! I think it should be good. Everything was added by the `cargo fmt` command, I just reverted the `;`

**Comment by albertov19** [comment]
Ah apologies I did not understand you wanted me to revert the `|` change as well - This should be ready to go right?

I tested this implementation in a local Foundry Forge instance and it worked btw.

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**